frontend/src/app/chatmenu/BlockUser.tsx

```diff
@@ -1,66 +1,79 @@
-import React from 'react'
+import React, { useState } from 'react'
 import './BlockUser.css'
-import '../ui/User.css'
 
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
-import { faUnlock, faUserLock } from '@fortawesome/free-solid-svg-icons'
+import { faLock, faLockOpen } from '@fortawesome/free-solid-svg-icons'
 
 import { ChatMenuPageProps } from './ChatMenuPage'
 import { UserData } from '../types/UserData'
+import { Button, UserTile } from '../ui'
+import { fetchFromBackend, useBackendFetchJson } from '../utils'
 
 interface BlockUserProps {
     participants: UserData[]
+    myUid: string
+}
+
+function FetchBlockedUsers(myUid: string, blockUid: string) {
+    const [res] = useBackendFetchJson({
+        route: `/chat/block/user/${myUid}`,
+        deps: [myUid],
+        isValid: myUid,
+    })
+    const blockedUsers = res?.blocks || []
+
+    return blockedUsers.find((user: { uid: string }) => user?.uid === blockUid)?.blocked || false
 }
 
 // create user tile to display participants
-const UserTile = ({ user }) => {
-    const { uid, displayName, photoURL } = user
-    const alt = `Profile image for ${displayName}.`
+const CreateUserTile = ({ user, myUid }) => {
+    const blockUid = user?.uid
+    const [isActive, setActive] = useState(FetchBlockedUsers(myUid, blockUid))
```

Comment:
I think we should move this state out of the `CreateUserTile` functional component, otherwise we will send a request each time we render a user. Instead, we should fetch the information at the top-level component, then pass in whether or not the user is blocked into the user element.